### PR TITLE
Optimize empty string compares

### PR DESCRIPTION
**Pull request by dawedawe**
## Description

According to [CA1820](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1820) it's more performant to do checks of empty strings by using `String.IsNullOrEmpty()` or the `Length` property.

## Checklist

- [ ] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [ ] Release notes entry updated:
    > Please make sure to add an entry with short succint description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Examples of release notes entries:
    > - Respect line limit in quick info popup - https://github.com/dotnet/fsharp/pull/16208
    > - More inlines for Result module - https://github.com/dotnet/fsharp/pull/16106
    > - Miscellaneous fixes to parens analysis - https://github.com/dotnet/fsharp/pull/16262
    >

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

File: src/Compiler/Driver/CreateILModule.fs [hunk 1]
```diff
@@ -584,19 +584,22 @@ module MainModuleBuilder =
                 [ resource ]
 
         // a user cannot specify both win32res and win32manifest
-        if not (tcConfig.win32manifest = "") && not (tcConfig.win32res = "") then
+        if
+            not (tcConfig.win32manifest.Length = 0)
```
Comment (psfinaki) [inline src/Compiler/Driver/CreateILModule.fs:588]: Shouldn't this also then be compared via IsNullOrEmpty?

Comment (psfinaki) [reply src/Compiler/Driver/CreateILModule.fs:588]: Also applies to a few other places in the PR where x.Length is used. 

The rules says that 
> Testing for null does not significantly affect the relative performance

hence it's probably worth just putting it everywhere like this as opposed to risking any NREs in the compiler code...

Comment (dawedawe) [reply src/Compiler/Driver/CreateILModule.fs]: One could argue that it's not strictly needed in these cases but yeah, let's be on the safe side and consistent :)

**Review by psfinaki — APPROVED** [review]
Thanks!

File: src/Compiler/Driver/CreateILModule.fs (hunk 1 above)
Comment (psfinaki) [reply src/Compiler/Driver/CreateILModule.fs]: Yeah it's also for cases when someone later decides to copypaste this code to a less safe context or add some logic in between... You know, paranoid mood today.

**Pull request merged**